spec.emu

```diff
@@ -47,24 +27,60 @@ contributors: Jordan Harband
-            1. Else if _toEscape_ contains _c_ or _c_ is matched by |WhiteSpace|, then
-              1. Append code unit U+005C (REVERSE SOLIDUS) to _escapedList_.
-            1. Append _c_ to _escapedList_.
+              1. Append code point U+005C (REVERSE SOLIDUS) to _escapedList_.
+              1. Append code point U+0078 (LATIN SMALL LETTER X) to _escapedList_.
+              1. Append code point U+0033 (DIGIT THREE) to _escapedList_.
+              1. Append _c_ to _escapedList_.
+            1. Else,
+              1. Append the code points in EncodeForRegExpEscape(_c_) to _escapedList_.
           1. Return CodePointsToString(_escapedList_).
         </emu-alg>
 
         <emu-note>
           <p>`escape` takes a string and escapes it so it can be literally represented as a pattern. In contrast EscapeRegExpPattern (as the name implies) takes a pattern and escapes it so that it can be represented as a string. While the two are related, they do not share the same character escape set or perform similar actions.</p>
         </emu-note>
       </emu-clause>
+
+      <emu-clause id="sec-encode" type="abstract operation">
+        <h1>
+          EncodeForRegExpEscape (
+            _c_: a code point,
+          ): a List of code points
+        </h1>
+        <dl class="header">
+          <dt>description</dt>
+          <dd>If the code unit represents a RegExp punctuator that needs escaping, or ASCII whitespace, it produces the code units for *"\x"* followed by the relevant escape code. If the code unit represents non-ASCII white space, it produces the code units for *"\u"* followed by the relevant escape code. Otherwise, it returns a List containing the original code unit.</dd>
+        </dl>
+
+        <emu-alg>
+          1. Let _codePoints_ be a new empty List.
+          1. Let _punctuators_ be the following String, which consists of every ASCII punctuator except U+005F (LOW LINE): *"(){}[]|,.?\*+-^$=<>\/#&!%:;@~'"`"*.
+          1. Let _toEscape_ be StringToCodePoints(_punctuators_).
+          1. If _toEscape_ contains _c_ or _c_ is matched by |WhiteSpace|, then
+            1. Append code point U+005C (REVERSE SOLIDUS) to _codePoints_.
+            1. Let _hex_ be Number::toString(𝔽(_c_), 16).
+            1. If the length of _hex_ is 1 or 2, then
+              1. Set _hex_ to StringPad(_hex_, 2, *"0"*, ~start~).
+              1. Append code point U+0078 (LATIN SMALL LETTER X) to _codePoints_.
+              1. Append the code points in StringToCodePoints(_hex_) to _codePoints_.
+            1. Else if the length of _hex_ > 4, then
+              1. Append code point U+0075 (LATIN SMALL LETTER U) to _codePoints_.
+              1. Append code point U+007B (LEFT CURLY BRACKET) to _codePoints_.
+              1. Append the code points in StringToCodePoints(_hex_) to _codePoints_.
+              1. Append code point U+007D (RIGHT CURLY BRACKET) to _codePoints_.
+            1. Else,
```

Comment:
One nit: I find it easier to read if the cases are written as `1-2, 3-4, >4` instead of `1-2, >4, 2-3`. Also then you can drop the `Assert: The length of _hex_ is at most 4.` since it will be quite obvious.

Reply:
sure, sounds good